x-pack/solutions/security/plugins/security_solution/server/lib/siem_migrations/common/types.ts

```diff
@@ -22,11 +30,23 @@ export interface SiemMigrationsClientDependencies {
   telemetry: AnalyticsServiceSetup;
 }
 
-export interface SiemMigrationsCommonCreateClientParams {
+export interface SiemMigrationsCreateClientParams<
+  D extends SiemMigrationsClientDependencies = SiemMigrationsClientDependencies
+> {
   request: KibanaRequest;
   currentUser: AuthenticatedUser | null;
   spaceId: string;
-  dependencies: SiemMigrationsClientDependencies;
+  dependencies: D;
 }
 
 export type SiemMigrationsIndexNameProvider = () => Promise<string>;
+
+export type Stored<T extends object> = T & { id: string };
+
+// TODO: replace these with the schemas for the common properties of the migrations and items
+// TODO: move these to the security_solution/common/siem_migrations/types.ts
```

Comment:
@semd Why? At least for this PR they are not needed to be moved to `common`. Let me know if I can remove these comments.

Reply:
The initial idea was to create a base type from which these two extend, but anyway, this is also fine, you are right. Let's remove these TODOs